src/libraries/Microsoft.PowerFx.Core/Public/Values/CustomObjectValue.cs

```diff
@@ -23,6 +23,8 @@ public interface ICustomObject
 
         bool IsNumber { get; }
 
+        bool IsBoolean { get; }
+
         object ToObject();
```

Comment:
>object ToObject(); [](http://example.com/codeflow?start=8&length=18)

can we avoid this on the interface? It's ambiguous semantics and can be hard for hosts to implement. 
The JSon obj just returns the json str, which is probably not what's expected. 

It looks like this is just used for FormulaValue.ToObject(), which is just used for marshalling back to C#. 

Can we have a default impl based on the other methods on the interface?
What about:
 object ToObj(ICustomObject x) {
    if (x.IsNumber()) return x.GetDouble();
    if (x.IsString()) return x.GetString(); 
    ...
    throw new NotImplemented("can't marshal to ob");
}

Reply:
This will not work because we have no way of returning arrays/objects.